lib/internal/warn.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+// A process Warning is similar in structure to an Error except that they
+// are emitted by the process.on('warn') event rather than being thrown.
+
+const _message = Symbol('message');
+const _name = Symbol('name');
+const prefix = `(${process.release.name}:${process.pid}) `;
+
+function warning(name, message, constructor) {
+  const err = new Error(`${prefix}${message}`);
+  err.name = name;
+  Error.captureStackTrace(err, constructor || warning);
```

Comment:
Yes, but calling Error.captureStackTrace with the constructor argument trims the stack trace to the actual call site as opposed to the line of code inside internal/warn.js.